react/components/ProductAssemblyContext/index.ts

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext, Dispatch } from 'react'
+
+export type DispatchAction = {
+  type: 'SET_QUANTITY'
+  args: {
+    itemId: string
+    newQuantity: number
+    type: string
+    groupPath: string[]
+  }
+}
+
+interface ItemState {
+  groupMaxQuantity: number
+  groupMinQuantity: number
+  item: AssemblyItem
+  groupType: string
+  groupQuantitySum: number
+  groupPath: string[]
+}
+
+export const ProductAssemblyDispatchContext = createContext<
+  Dispatch<DispatchAction>
+>(() => {})
+export const useProductAssemblyDispatch = () =>
```

Comment:
Is that a dispatch? Where is the useReducer?

Reply:
use reducer is on StateManager